internal/pkg/blockdevice/probe/probe.go

```diff
@@ -150,6 +87,97 @@ func GetDevWithFileSystemLabel(value string) (probe *ProbedBlockDevice, err erro
 		return nil, err
 	}
 
+	return filterByLabel(probed, value)
+}
+
+func ProbeDevForFileSystemLabel(devpath, value string) (probe *ProbedBlockDevice, err error) {
+	var probed []*ProbedBlockDevice
+	probed, err = probeByFilesystem(devpath)
+	if err != nil {
+		return nil, err
+	}
+
+	probe, err = filterByLabel(probed, value)
+	if err != nil {
+		return nil, err
+	}
+
+	return probe, err
+}
+
+func probe(devpath string) (devpaths []string) {
+	devpaths = []string{}
+
+	// Start by opening the block device.
+	// If a partition table was not found, it is still possible that a
+	// file system exists without a partition table.
+	bd, err := blockdevice.Open(devpath)
+	if err != nil {
+		// nolint: errcheck
+		if sb, _ := FileSystem(devpath); sb != nil {
+			devpaths = append(devpaths, devpath)
+		}
+
+		return devpaths
+	}
+	// nolint: errcheck
+	defer bd.Close()
+
+	// A partition table was not found, and we have already checked for
+	// a file system on the block device. Let's check if the block device
+	// has partitions.
+	pt, err := bd.PartitionTable(true)
+	if err != nil {
+		return devpaths
+	}
+
+	// A partition table was found, now probe each partition's file system.
+	name := filepath.Base(devpath)
+	for _, p := range pt.Partitions() {
+		var partpath string
+		switch {
+		case strings.HasPrefix(name, "nvme"):
+			fallthrough
+		case strings.HasPrefix(name, "loop"):
+			partpath = fmt.Sprintf("/dev/%sp%d", name, p.No())
+		default:
+			partpath = fmt.Sprintf("/dev/%s%d", name, p.No())
+		}
+		// nolint: errcheck
+		if sb, _ := FileSystem(partpath); sb != nil {
+			devpaths = append(devpaths, partpath)
+		}
+	}
+
+	return devpaths
+}
+
+func probeByFilesystem(devpath string) (probed []*ProbedBlockDevice, err error) {
```

Comment:
`probeFilesystem` is better

Reply:
`probeDevice` in this case seems more appropriate.

Reply:
Or maybe this should be `probe` and the existing `probe` should be renamed to `probeDevice`?  #namingishard

Reply:
I imagine in the future we can have probing funcs that probe specific attributes. In this ase this function probes for a filesystem, and errors if it can't find one.